src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -93,8 +93,7 @@ public Resource Merge(Resource other)
             string sanitizedKey;
             if (attribute.Key == null)
             {
-                OpenTelemetrySdkEventSource.Log.InvalidArgument("Create resource", "attribute key", "Attribute key should be non-null string.");
-                sanitizedKey = string.Empty;
+                throw new System.ArgumentException("Resource's attributes contains a null key");
```

Comment:
In response to:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1720/files#r564877372

Reply:
The message might be misleading since it is talking about "attributes on the resource" but the method is only focusing on one attribute.
One possible way is to put something like `nameof(attribute) has a null key` here.

Reply:
I might be misunderstanding but each attribute is just a key and value, so a null key makes it impossible to do that right? I could print the Value instead since Key is null, but it may also be null and would then require checking and formatting

Reply:
Found out C# prevents you from instantiating any dictionary where the key is null, it'll throw its own ArgumentNullException. Might be the reason we originally didn't check for key == null, and I'm happy to just remove this change if that makes sense.